### ISSUE TITLE
Question regarding the "weight" option in did_multiplegt_dyn

### ISSUE THREAD

**Issue by qy-han** [chaisemartinPackages/did_multiplegt_dyn#44]
Recently we were trying to use the “weight” option but we encountered some technical issues.
 
The weighting variable, named “cem_weight”, of each unit in our data were calculated using the “coarsened exact matching” method to facilitate a matched DID design. We subsequently used the following command:
 
**did_multiplegt_dyn outcome nstate_c year treat, weight(cem_weight) effects($Ndyneffect) placebo($Nplacebo) only_never_switchers**
 
but Stata aborts with an error message that the variable “cem_weight” cannot be found. We did check that this variable is in our dataset, so we would like to seek your advice on what might be wrong in our code.
 
Thank you very much for your help!

**Comment by chaisemartinPackages** [comment]
Dear Han,
This is Diego from Clément de Chaisemartin's RA team.
Thanks for your interest in did_multiplegt_dyn!
I have tried running cem and did_multiplegt_dyn in a very basic scenario and I found no error:

```stata
qui bcuse wagepan
cem black hours married, treatment(union)
did_multiplegt_dyn lwage nr year union, weight(cem_weight) only_never_switchers
```

To this end, I fear that this may be related to something in your data or in the code before the command.
It would be great if you could send us a reproducible example where this error shows up.
Here's out team mail account: [chaisemartin.packages@gmail.com](mailto:chaisemartin.packages@gmail.com)
Best,
Diego

**Comment by qy-han** [comment]
Hi Diego,

Thank you very much for your explanation! This is super helpful. I have made a few tests on my dataset. Comparing with structures between my data and your sample data, I found that the error message "cem_weight not found" pops out because our CEM weight varies within each (state, year) cell, as we are using a individual-level data. But in using did_multiplegt_dyn, we use state as the group indicator. Once I changed to state-level data, this error message disappears. 

May I ask if there is a function with the command that deals with this kind of situation? i.e. treatment may happen at a more aggregated level, but the data and the weight is at individual level. 

Thank you!!
